src/apps/activity_assignments/service.py

```diff
@@ -234,6 +239,54 @@ async def get_all(self, applet_id: uuid.UUID, query_params: QueryParams) -> list
             for schema in schemas
         ]
 
+    async def get_all_by_respondent(
+        self, applet_id: uuid.UUID, respondent_subject_id: uuid.UUID
+    ) -> list[ActivityAssignmentWithSubject]:
+        schemas = await ActivityAssigmentCRUD(self.session).get_by_applet_and_respondent(
+            applet_id, respondent_subject_id
+        )
+
+        respondent_subject = await SubjectsCrud(self.session).get_by_id(respondent_subject_id)
+        if not respondent_subject:
+            raise NotFoundError()
+
+        target_subject_ids = [schema.target_subject_id for schema in schemas]
+
+        target_subjects = {
+            subject.id: SubjectReadResponse(
+                id=subject.id,
+                first_name=subject.first_name,
+                last_name=subject.last_name,
+                email=subject.email,
+                language=subject.language,
+                nickname=subject.nickname,
+                secret_user_id=subject.secret_user_id,
+                tag=subject.tag,
+                applet_id=subject.applet_id,
+            )
+            for subject in await SubjectsCrud(self.session).get_by_ids(target_subject_ids)
```

Comment:
Since we throw `NotFoundError()` for the respondent subject, should we also throw it for any of the target subjects in case they're not found? And if so, maybe the `NotFoundError` should include a message with context (e.g. the subject ID)

Reply:
done!